core/rawdb/accessors_l1_message.go

```diff
@@ -4,14 +4,24 @@ import (
 	"bytes"
 	"encoding/binary"
 	"math/big"
+	"time"
 
 	"github.com/scroll-tech/go-ethereum/common"
 	"github.com/scroll-tech/go-ethereum/core/types"
 	"github.com/scroll-tech/go-ethereum/ethdb"
 	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/metrics"
 	"github.com/scroll-tech/go-ethereum/rlp"
 )
 
+var (
+	// L1 message iterator metrics
+	iteratorNextCalledCounter      = metrics.NewRegisteredCounter("rawdb/l1_message/iterator/next_called", nil)
+	iteratorInnerNextCalledCounter = metrics.NewRegisteredCounter("rawdb/l1_message/iterator/inner_next_called", nil)
+	iteratorLengthMismatchCounter  = metrics.NewRegisteredCounter("rawdb/l1_message/iterator/length_mismatch", nil)
+	iteratorNextDurationTimer      = metrics.NewRegisteredTimer("rawdb/l1_message/iterator/next_time", nil)
```

Comment:
iteratorL1MessageSizeGuage = metrics.NewRegisteredGauge("rawdb/l1_message/size", nil)

Reply:
add this at https://github.com/scroll-tech/go-ethereum/pull/527/files#diff-8ae078aff548d4b673c0f691e95a0085fb41a9f0e4e5a9798fde19e2be797d2eR226

```go
iteratorL1MessageSizeGauge.Update(unsafe.Sizeof(msg))
```